drivers: clock_control: stm32: hide STM32_HSE_CLOCK when HSE is off

mathieuchopstm opened the pull request:
Ensure the `CONFIG_CLOCK_STM32_HSE_CLOCK` symbol is not visible when the HSE clock is not enabled.

Depends on #97215.

mathieuchopstm committed:
drivers: clock_control: stm32: define STM32_HSE_CLOCK based only on DT

Ensure the CONFIG_CLOCK_STM32_HSE_CLOCK symbol cannot be overridden by
making it depend on the Device Tree node "clk_hse" being enabled. The
symbol's value is also directly sourced from that node instead of being
user-configurable.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

erwango reviewed (changes requested):
To be documented in migration guide for MP based platforms ?

Could also be rebased now that https://github.com/zephyrproject-rtos/zephyr/pull/97215 has been merged

arnopo commented:
> > maybe better solution: make HSE node mandatory in DT but take its value even if status = "disabled";
> 
> I'm actually fine with implemented proposal. This avoids having a specific behavior for a specific node (which can be tricky to guess)

we could have same aproach than stm32mp2 with status = "okay"

---------------- boards/st/stm32mp157c_dk2/Kconfig.defconfig -----------------
```diff
index 40490cb49c3f..1f58590c079d 100644
@@ -9,7 +9,4 @@ config SPI_STM32_INTERRUPT
 	default y
 	depends on SPI
 
-config CLOCK_STM32_HSE_CLOCK
-	default 24000000
-
 endif # BOARD_STM32MP157_Dk2
```

---------------- boards/st/stm32mp157c_dk2/stm32mp157c_dk2.dts ----------------
```diff
index b5057ebbfd2d..07cce08fda09 100644
@@ -57,6 +57,11 @@
 	clock-frequency = <DT_FREQ_M(209)>;
 };
 
+&clk_hse {
+	clock-frequency = <DT_FREQ_M(32)>;
+	status = "okay";
+};
+
 &spi4_miso_pe13 {
 	slew-rate = "very-high-speed";
 };
```

------------------------ dts/arm/st/mp1/stm32mp157.dtsi ------------------------
```diff
index eda7922f65a4..477b243d6295 100644
@@ -412,6 +412,21 @@
 		i2c = <&i2c5>;
 		status = "disabled";
 	};
+
+	clocks {
+		clk_hse: clk-hse {
+			#clock-cells = <0>;
+			compatible = "fixed-clock";
+			status = "disabled";
+		};
+
+		clk_hsi: clk-hsi {
+			#clock-cells = <0>;
+			compatible = "fixed-clock";
+			clock-frequency = <DT_FREQ_M(64)>;
+			status = "disabled";
+		};
+	};
 };
```
I tested with command `west build -b stm32mp157c_dk2 samples/subsys/ipc/openamp_rsc_table`
as result "HSE_VALUE=24000000" as expected

mathieuchopstm committed:
dts: arm: st: mp1: add HSE clock to DTSI

Add the HSE clock to STM32MP1 series DTSI.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: st: stm32mp157c_dk2: provide HSE clock frequency in DTS

Provide the HSE clock frequency by enabling and configuring its Device Tree
node in the board's DTS instead of using Kconfig.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: oct: osd32mp1_brk: provide HSE clock frequency in DTS

Provide the HSE clock frequency by enabling and configuring its Device Tree
node in the board's DTS instead of using Kconfig.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

etienne-lms reviewed (dismissed):
All good to me.

etienne-lms commented:
Remains @erwango's comment about adding a word in the migration guide (https://github.com/zephyrproject-rtos/zephyr/pull/97256#pullrequestreview-3339256769).

mathieuchopstm committed:
doc: migration_guide: 4.3: add note about CONFIG_CLOCK_STM32_HSE_CLOCK

Document that CONFIG_CLOCK_STM32_HSE_CLOCK is no longer user-configurable
and the replacement mechanism for setting its value (through Device Tree).

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>
